pkg/tagger/remote/tagger.go

```diff
@@ -215,8 +215,10 @@ func (t *Tagger) run() {
 	for {
 		select {
 		case <-t.health.C:
+			continue
```

Comment:
I guess it's on purpose, but the `continue` skips the remaining of the `for` loop entirely, is it expected?

Reply:
Yes, it's expected. It forces this select to be re-evaluated. If not, a successful `<-t.health.C` will skip a check on `t.ctx.Done`, which may result in a new connection being made instead of an early return.